src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc

```diff
@@ -33,184 +33,103 @@
+ public:
+  AddSubMenuRequestTest()
+      : mock_message_helper_(*MockMessageHelper::message_helper_mock()) {}
+  MockMessageHelper& mock_message_helper_;
+};
+
+TEST_F(AddSubMenuRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+  const uint32_t menu_id = 10u;
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  (*msg)[am::strings::msg_params][am::strings::menu_id] = menu_id;
+
+  utils::SharedPtr<AddSubMenuRequest> command =
+      CreateCommand<AddSubMenuRequest>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(kConnectionKey));
+  EXPECT_CALL(*mock_app, AddSubMenu(menu_id, _));
+  EXPECT_CALL(*mock_app, UpdateHash());
+
+  MessageSharedPtr ev_msg = CreateMessage(smart_objects::SmartType_Map);
+  (*ev_msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*ev_msg)[am::strings::msg_params][am::strings::info] = "info";
 
-  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
-
-  command->on_event(event);
-}
-
-TEST_F(AddSubMenuRequestTest, OnEvent_SUCCESS) {
   Event event(hmi_apis::FunctionID::UI_AddSubMenu);
-  MessageSharedPtr event_msg(CreateMessage(smart_objects::SmartType_Map));
-  (*event_msg)[am::strings::params][am::hmi_response::code] =
-      mobile_apis::Result::SUCCESS;
-  (*event_msg)[am::strings::msg_params] = 0;
+  event.set_smart_object(*ev_msg);
 
-  event.set_smart_object(*event_msg);
+  MockHmiInterfaces hmi_interfaces;
+  ON_CALL(app_mngr_, hmi_interfaces()).WillByDefault(ReturnRef(hmi_interfaces));
+  EXPECT_CALL(hmi_interfaces,
+              GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI))
+      .WillOnce(Return(am::HmiInterfaces::STATE_AVAILABLE));
 
-  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
-  (*command_msg)[am::strings::params][am::strings::connection_key] =
-      kConnectionKey;
+  EXPECT_CALL(mock_message_helper_,
+              HMIToMobileResult(hmi_apis::Common_Result::UNSUPPORTED_RESOURCE))
+      .WillOnce(Return(mobile_apis::Result::UNSUPPORTED_RESOURCE));
 
-  AddSubMenuPtr command(CreateCommand<AddSubMenuRequest>(command_msg));
-
-  MockAppPtr app(CreateMockApp());
-  EXPECT_CALL(app_mngr_, application(kConnectionKey)).WillOnce(Return(app));
-
-  MessageSharedPtr result_msg(
-      CatchMobileCommandResult(CallOnEvent(*command, event)));
-  const mobile_apis::Result::eType kReceivedResult =
-      static_cast<mobile_apis::Result::eType>(
-          (*result_msg)[am::strings::msg_params][am::strings::result_code]
-              .asInt());
-  EXPECT_EQ(mobile_apis::Result::SUCCESS, kReceivedResult);
-}
+  MessageSharedPtr ui_command_result;
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
```

Comment:
@okozlovlux in my PR you asked to check everywhere both args in ManageMobileCommand(..), so shouldn't be the same here?